pkg/virt-api/rest/expand.go

```diff
@@ -101,12 +103,25 @@ func expandSpecResponse(vm *v1.VirtualMachine, instancetypeMethods instancetype.
 		return
 	}
 
-	conflicts := instancetypeMethods.ApplyToVmi(field.NewPath("spec", "template", "spec"), instancetypeSpec, preferenceSpec, &vm.Spec.Template.Spec, &vm.Spec.Template.ObjectMeta)
+	util.SetDefaultVolumeDisk(&vm.Spec.Template.Spec)
+
+	if err := vmispec.SetDefaultNetworkInterface(app.clusterConfig, &vm.Spec.Template.Spec); err != nil {
+		writeError(errorFunc(err), response)
+		return
+	}
```

Comment:
Q: Could we reuse dry-run create?

Reply:
@xpivarc  Unfortunately not, we only apply instance types and preferences through the VirtualMachine controller. Even reusing dry-run create for these defaults wouldn't work as others would also be set (vCPU topology, resources etc) that would conflict with any provided the instance type. 

So unfortunately we end up duplicating most of the logic from the controller here. I do want to clean this up in a follow up series once this change has landed and been backported.

Reply:
>So unfortunately we end up duplicating most of the logic from the controller here. I do want to clean this up in a follow up series once this change has landed and been backported.

Looking forward